### PR TITLE
[WIP] Include version in exceptions

### PR DESCRIPTION
**Pull request by kneth**
This pull request introduces two things:

* Version to exception thrown at JNI level. 
* A utility to set version (`tools/set-version.sh`)

Setting the version for a new release requires that `tools/set-version.sh` and `realm-jni/generate-jni-headers.sh` to be run.

@emanuelez @cmelchior @bmunkholm

### CODE REVIEW

File: tools/set-version.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+if [ -z "$1" ]; then
+    echo "You must supply a version number"
```
Comment (cmelchior) [inline tools/set-version.sh]: Describe what a valid number looks like.

File: realm-jni/src/io_realm_internal_Util.cpp [hunk 2]
```diff
@@ -127,6 +127,9 @@ JNIEXPORT jstring JNICALL Java_io_realm_internal_Util_nativeTestcase(
     if (dotest) {
         return NULL;
     }
-    return to_jstring(env, expect);
+    string version = "Realm version " + num_to_string<int>(io_realm_internal_Version_REALM_JAVA_MAJOR) + "."
```
Comment (cmelchior) [inline realm-jni/src/io_realm_internal_Util.cpp]: Debatting with myself if it is too verbose having the "Realm version". Wouldn't just the version number be enough as it is for people copy / pasting errors to us?

File: tools/set-version.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    exit 1
+fi
+
+realm_java_version="$1"
+realm_java_version_file="realm/src/main/java/io/realm/internal/Version.java"
+realm_java_ver_major="$(echo "$realm_java_version" | cut -f1 -d.)" || exit 1
```
Comment (cmelchior) [inline tools/set-version.sh]: Why not just use ```set -e``` at the top of the file?

File: realm-jni/src/io_realm_internal_Util.cpp (hunk 2 above)
Comment (bmunkholm) [inline realm-jni/src/io_realm_internal_Util.cpp]: maybe "realm-java-VERSION"

**Comment by emanuelez** [comment]
I think this is not the best way to do this. We should only update version.txt and everything else should be handled by gradle

**Comment by bmunkholm** [comment]
@kneth: Should this be closed

**Comment by kneth** [comment]
@bm: Yes, it looks that we never really wished to add the version number.

**Pull request closed without merging**